src/xrpl-account.js

```diff
@@ -99,26 +99,45 @@ class XrplAccount {
         return xrpl.parseAccountRootFlags((await this.getInfo()).Flags);
     }
 
-    async setMessageKey(publicKey, options = {}) {
-        const result = await this.#submitAndVerifyTransaction({
-            TransactionType: 'AccountSet',
-            Account: this.address,
-            MessageKey: publicKey,
-        }, options);
-        return result;
-    }
 
-    setDefaultRippling(enabled, options = {}) {
+
+    setAccountFields(fields, options = {}) {
+        /**
+         * Example for fields
+         * 
+         * fields = {
+         *  Domain : "www.mydomain.com",
+         *  Flags : { asfDefaultRipple: false, asfDisableMaster: true } 
+         * }
+         * 
+         */
+
+        if (Object.keys(fields).length === 0)
+            throw "AccountSet fields cannot be empty.";
 
         const tx = {
             TransactionType: 'AccountSet',
             Account: this.address
-        }
+        };
+
+        for (const [key, value] of Object.entries(fields)) {
+
+            switch (key) {
+                case 'Domain' :
+                    tx.Domain = TransactionHelper.asciiToHex(value).toUpperCase();
+                    break;
 
-        if (enabled)
-            tx.SetFlag = xrpl.AccountSetAsfFlags.asfDefaultRipple;
-        else
-            tx.ClearFlag = xrpl.AccountSetAsfFlags.asfDefaultRipple;
+                case 'Flags' :
+                    for (const [flagKey, flagValue] of Object.entries(value)) {
+                        tx[(flagValue) ? 'SetFlag' : 'ClearFlag'] = xrpl.AccountSetAsfFlags[flagKey];
```

Comment:
Flags can be combined. This logic will always reset the current flag value. Instead you need to 'bitwise OR' the flag key value with the existing SetFlag or ClearFlag value.